pkg/cloudprovider/providers/vsphere/vclib/datastore.go

```diff
@@ -47,3 +47,23 @@ func (ds *Datastore) GetType(ctx context.Context) (string, error) {
 	}
 	return dsMo.Summary.Type, nil
 }
+
+// IsCompatibleWithStoragePolicy returns true if datastore is compatible with given storage policy else return false
+func (ds *Datastore) IsCompatibleWithStoragePolicy(ctx context.Context, storagePolicyId string) (bool, error) {
+	pbmClient, err := NewPbmClient(ctx, ds.Client())
+	if err != nil {
+		glog.Errorf("Failed to get new PbmClient Object. err: %v", err)
+		return false, err
+	}
+	datastores := []*Datastore{ds}
+	compatibleDatastore, err := pbmClient.GetCompatibleDatastores(ctx, storagePolicyId, datastores)
+	if err != nil {
+		glog.Errorf("Failed to get match compatible datastore for storage policy id: %s. err: %v", storagePolicyId, err)
+		return false, err
+	}
+	if compatibleDatastore[0].Datastore.Reference() == ds.Datastore.Reference() {
```

Comment:
This returns only one datastore or above error, since we are passing only one datastore in the list to check for.

Will be used as 

```
compatible, err := datastore.IsCompatibleWithStoragePolicy(ctx, vmdisk.volumeOptions.StoragePolicyID)

if err != nil {
			glog.Errorf("Error Occurred while checking datastore compatibility with storage policy id: %s, err: %+v", vmdisk.volumeOptions.StoragePolicyID, err)
			return err
		}
```